prov/usnic/src/usdf_ep_msg.c

```diff
@@ -106,6 +108,7 @@ usdf_tx_msg_enable(struct usdf_tx *tx)
 			sizeof(struct usdf_msg_qe));
 	if (tx->t.msg.tx_wqe_buf == NULL) {
 		ret = -errno;
+		USDF_INFO("malloc failed (%s)\n", strerror(-ret));
```

Comment:
Nope, this was deliberate.  If the `USDF_INFO` macro is implemented in a way that it calls a function that twiddles `errno` before it actually evaluates the `strerror(errno)`, we'll get the wrong result.  My general rule of thumb w.r.t. `errno` is to sample it as close as possible to the call that set it and then don't attempt to use it again.

In this particular instance, it would probably be fine to make the change you are suggesting.  But I don't see anything confusing about it as is and I prefer the more defensive version.